src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -291,9 +290,8 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       priorityContext = nextPriorityLevel;
 
       if (
-        didCommit &&
-        (nextPriorityLevel === TaskPriority ||
-          nextPriorityLevel === SynchronousPriority)
+        nextPriorityLevel === TaskPriority ||
+        nextPriorityLevel === SynchronousPriority
```

Comment:
My reasoning is that it's always ok to do an async update in `componentDidUpdate` because it's non-blocking. So we shouldn't ever throw this error for an async update. E.g. If you do LIMIT - 1 nested sync updates, and then an additional async update, that should be fine.